### PR TITLE
refactor: pytorch multiargs better support and partial_kwargs impl

### PR DESCRIPTION
**Pull request by larme**
This is more like a proposal for adding partial_kwargs to pytorch.

The main motivation is that pytorch's `forward` is quite flexible and may accept arbitrary arguments. For a runner, sometimes all input share the same value for a particular argument (most likely a configuration parameter) and in this case it's better to specify it at load_runner stage.

Will add docs and tests later.

## Checklist:

- [ ] My code follows the bentoml code style, both `./dev/format.sh` and
  `./dev/lint.sh` script have passed
  ([instructions](https://github.com/bentoml/BentoML/blob/master/DEVELOPMENT.md#style-check-and-auto-formatting-your-code)).
- [ ] My change reduces project test coverage and requires unit tests to be added
- [ ] I have added unit tests covering my code change
- [ ] My change requires a change to the documentation
- [ ] I have updated the documentation accordingly

### CODE REVIEW

File: bentoml/pytorch.py [hunk 1]
```diff
@@ -230,28 +232,33 @@ def _setup(self) -> None:  # type: ignore[override]
     @torch.no_grad()
     def _run_batch(  # type: ignore[override]
```
Comment (aarnphm) [inline bentoml/pytorch.py:233]: I don't think we need this `# type: ignore` anymore
```suggestion
    def _run_batch(
```

File: bentoml/pytorch.py [hunk 2]
```diff
@@ -230,28 +232,33 @@ def _setup(self) -> None:  # type: ignore[override]
...
+                if isinstance(item, torch.Tensor):
+                    item = item.cuda()
 
-        res: "torch.Tensor"
+            return item
+
+        params = params.map(_mapping)
+
+        # merge kwargs with partial_kwargs
+        merged_kwargs = {**self._partial_kwargs, **kwargs}
```
Comment (aarnphm) [inline bentoml/pytorch.py:254]: any updates from our previous chat? @larme

Comment (larme) [reply bentoml/pytorch.py]: @aarnphm done, just wanted to finish the test then do this so the refactoring is tested~

File: bentoml/pytorch.py (hunk 1 above)
Comment (larme) [reply bentoml/pytorch.py:234]: let me correct this part for all frameworks in another commit

File: bentoml/pytorch.py (hunk 2 above)
Comment (aarnphm) [reply bentoml/pytorch.py]: i saw the push, well done 👍🏼

**Pull request merged**